tests/plugins/widget/editing.js

```diff
@@ -55,6 +55,31 @@
 							]
 						};
 					} );
+
+					evt.editor.widgets.add( 'dialogtest1', {
+						template: '<div>bar</div>',
+						dialog: 'dialogtest1'
+					} );
+
+					CKEDITOR.dialog.add( 'dialogtest1', function() {
+						return {
+							title: 'DialogTest1',
+							contents: [
+								{
+									id: 'info',
+									elements: []
+								}
+							],
+							onShow: dialogEventListener,
+							onOk: dialogEventListener,
+							onHide: dialogEventListener
+						};
+					} );
+
+					function dialogEventListener( evt ) {
+						assert.isNotUndefined( this.widget, 'No widget passed for: ' + evt.name );
+						assert.areEqual( 'dialogtest1', this.widget.name, 'Invalid widget for: ' + evt.name );
```

Comment:
If the assertions were inside the test, the whole listener would just check if `this.widget === widget`.